### PR TITLE
Fvr check for duplicate emis

### PR DESCRIPTION
**Pull request by frankvanrietimqs**
The basic problem is that when we update an asset with a different EMIS number a new asset-grouping entry is made without deleting the previous one, so it is impossible to change an emis during an import.  The asset grouping mechanism is supposed to be a general one though so one does not want to build too many rules into the server.

Compromise implemented: When importing EMIS (the very specific type of asset grouping) all existing entries for that asset and that grouping type are deleted and then new one then added.  This is achieved by adding a general API call to the back-end but adding client specific logic to the importer. This is in keeping with our approach to have client specific importers.

As a safety net there is also a client specific constraint added.  This is more contentious. Hiding these things in code make maintenance more difficult and surprises people. It would be better if we could have it in the client specfic liquibase schema. **_I am open to suggestions on how to do this better._**

### CODE REVIEW

File: Service/src/test/java/za/co/imqs/api/asset/AbstractAssetControllerAPITest.java [hunk 1]
```diff
@@ -360,31 +360,42 @@ public static Timestamp fetchAssetLastChanged(UUID uuid) {
     public static void assertCorrectlyUpdated(Timestamp then, Timestamp now) {
         Assert.assertNotNull(then);
         Assert.assertNotNull(now);
-        Assert.assertTrue(now.compareTo(then) > 0 && now.getTime() - then.getTime() < 1000);
+        Assert.assertTrue(now.compareTo(then) > 0 && Math.abs(now.getTime() - then.getTime()) < 1000);
```
Comment (CalebBredekamp) [inline Service/src/test/java/za/co/imqs/api/asset/AbstractAssetControllerAPITest.java:363]: Not sure if this matters much, but doesn't this mean that `then` is now technically be allowed to be ahead of `now`?

**Review by CalebBredekamp — APPROVED** [review]
Cool, I'm with you on the rest of it

File: Service/src/test/java/za/co/imqs/api/asset/AbstractAssetControllerAPITest.java (hunk 1 above)
Comment (frankvanrietimqs) [reply Service/src/test/java/za/co/imqs/api/asset/AbstractAssetControllerAPITest.java:363]: Sure. This is just force of habit in dealing with integer values wrapping. Lets assume this system will still be running in 2038 ... :)

Comment (CalebBredekamp) [reply Service/src/test/java/za/co/imqs/api/asset/AbstractAssetControllerAPITest.java:363]: Haha sure okay

**Pull request merged**